### PR TITLE
[WIP] Refactor/open pack state

### PR DESCRIPTION
**Pull request by rilifon**
Refactoring open pack screen to display all cards in the pack, and player can choose which ones he wants to consume.

After confirming, all remaining cards will be add to player's buffer

### CODE REVIEW

File: game/domain/definitions/init.lua [hunk 1]
```diff
@@ -12,10 +12,9 @@ DEFS.WIDGETS = {
   "WIDGET_C",
   "WIDGET_D",
 }
-DEFS.ACTOR_BUFFER_NUM = 3
-DEFS.PACK_SIZE = 10
+DEFS.ACTOR_BUFFER_NUM = 1
```
Comment (OrenjiAkira) [inline game/domain/definitions/init.lua]: Remove this line

File: game/gamestates/open_pack.lua [hunk 2]
```diff
@@ -34,28 +60,36 @@ local function _moveFocus(dir)
...
-end
-
-local function _confirmCard()
-  local action_type, buffer_index = _pack_view:getTarget()
+local function _consumeCard()
   table.insert(_picks, {
-    action_type = action_type,
-    buffer_index = buffer_index,
+    action_type = "consume",
+    buffer_index = 0,
```
Comment (OrenjiAkira) [inline game/gamestates/open_pack.lua]: no need for a buffer index, because there is only one now

File: game/gamestates/open_pack.lua [hunk 3]
```diff
@@ -34,28 +60,36 @@ local function _moveFocus(dir)
...
     SWITCHER.pop(_picks)
     _picks = nil
   end
 end
 
+local function _confirm()
+  while not _pack_view:isEmpty() do
+    table.insert(_picks, {
+      action_type = "get",
+      buffer_index = 1,
```
Comment (OrenjiAkira) [inline game/gamestates/open_pack.lua]: same here: remove buffer_index value

**Comment by OrenjiAkira** [comment]
Also please rebase this branch

**Comment by Kazuo256** [comment]
Fixed and rebased.

**Pull request merged**